lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -345,22 +345,26 @@ export default class DocumentComposer {
       }
 
       // `serviceEndpoint` validations.
-      const serviceEndpoint = service.serviceEndpoint;
-      if (typeof serviceEndpoint === 'string') {
-        const uri = URI.parse(service.serviceEndpoint);
-        if (uri.error !== undefined) {
-          throw new SidetreeError(
-            ErrorCode.DocumentComposerPatchServiceEndpointStringNotValidUri,
-            `Service endpoint string '${serviceEndpoint}' is not a valid URI.`
-          );
-        }
-      } else if (typeof serviceEndpoint === 'object') {
-        // Allow `object` type only if it is not an array.
-        if (Array.isArray(serviceEndpoint)) {
-          throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointCannotBeAnArray);
+      // transform URI strings and JSON objects into array so that we can run validations more easily
+      const serviceEndpointValueAsArray = Array.isArray(service.serviceEndpoint) ? service.serviceEndpoint : [service.serviceEndpoint];
+      for (const serviceEndpoint of serviceEndpointValueAsArray) {
+        // serviceEndpoint itself must be URI string or non-array object
+        if (typeof serviceEndpoint === 'string') {
+          const uri = URI.parse(serviceEndpoint);
+          if (uri.error !== undefined) {
+            throw new SidetreeError(
+              ErrorCode.DocumentComposerPatchServiceEndpointStringNotValidUri,
+              `Service endpoint string '${serviceEndpoint}' is not a valid URI.`
+            );
+          }
+        } else if (typeof serviceEndpoint === 'object') {
+          // Allow `object` type only if it is not an array.
+          if (Array.isArray(serviceEndpoint)) {
+            throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointCannotBeAnArray);
+          }
+        } else {
+          throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointMustBeStringOrNonArrayObject);
```

Comment:
We will need the same change in [ion-sdk](https://github.com/decentralized-identity/ion-sdk/blob/38f4cbea1fed7dacdc2dc63d9a346eeffe0b4e66/lib/IonRequest.ts#L328). Totally get it if you don't have time to make those changes in `ion-sdk`, so the PR will suffice to be approved if you can file an issue in the `ion-sdk` repo to make the same changes! Thanks!

Reply:
Happy to implement this in ion-sdk as well. I will still create an issue as well, as I am not yet 100% sure when I will find time for the implementation. So I will open the issue, once this PR is merged, right?

Reply:
Hi @thehenrytsai, I updated ion-sdk (decentralized-identity/ion-sdk#30) as well